src/upgrades/pre_0_10_0/pubd_migration.rs

```diff
@@ -0,0 +1,201 @@
+            // In this case IdCert -> IdCertInfo for added publishers. Then save the event
+            // again in the migration scope.
+            if let Some(event_versions) = cmd.effect().events() {
+                for v in event_versions {
+                    let event_key = Self::event_key(scope, *v);
+                    trace!("  +- event: {}", event_key);
+                    let evt: OldRepositoryAccessEvent = self
+                        .current_kv_store
+                        .get(&event_key)?
+                        .ok_or_else(|| PrepareUpgradeError::Custom(format!("Cannot parse old event: {}", event_key)))?;
+
+                    // Migrate into the current event type and save
+                    let evt: RepositoryAccessEvent = evt.into();
+                    self.new_kv_store.store(&event_key, &evt)?;
+                }
+            }
+
+            // Save the command to the migration
+            self.new_kv_store.store(&cmd_key, &cmd)?;
+
+            // Update and save data_upgrade_info for progress tracking
+            data_upgrade_info.last_command += 1;
+            data_upgrade_info.last_update = cmd.time();
+            self.update_data_upgrade_info(scope, &data_upgrade_info)?;
+
+            // Report progress and expected time to finish on every 100 commands evaluated.
+            total_migrated += 1;
+            if total_migrated % 100 == 0 {
+                // expected time: (total_migrated / (now - started)) * total
+
+                let mut time_passed = (Time::now().timestamp() - time_started.timestamp()) as usize;
+                if time_passed == 0 {
+                    time_passed = 1; // avoid divide by zero.. we are doing approximate estimates here
+                }
+                let migrated_per_second: f64 = total_migrated as f64 / time_passed as f64;
+                let expected_seconds = (total_commands as f64 / migrated_per_second) as i64;
+                let eta = time_started + Duration::seconds(expected_seconds);
+                info!(
+                    "  migrated {} commands, expect to finish: {}",
+                    total_migrated,
+                    eta.to_rfc3339()
+                );
+            }
+        }
+
+        info!("Finished migrating Publication Server commands");
+
+        // Create a new info file for the new aggregate repository
+        {
+            let info = StoredValueInfo::from(&data_upgrade_info);
+            let info_key = KeyStoreKey::scoped(scope.to_string(), "info.json".to_string());
+            self.new_kv_store.store(&info_key, &info)?;
+        }
+
+        // Verify migration
+        info!("Will verify the migration by rebuilding the Publication Server from events");
+        let repo_access = self.new_agg_store.get_latest(&handle).map_err(|e| {
+            PrepareUpgradeError::Custom(format!(
+                "Could not rebuild state after migrating pubd! Error was: {}.",
```

Comment:
You don't refer to `pubd` elsewhere, should this say "Publication Server" ?

Reply:
ack